plugins/modules/nd_api_key.py

```diff
@@ -0,0 +1,222 @@
+    api_key_name: "ansible_test_key"
+    annotations:
+      owner: "ansible"
+      purpose: "automation"
+    state: present
+  register: result
+
+- name: Create API key with minimal config
+  cisco.nd.nd_api_key:
+    api_key_name: "simple_key"
+    state: present
+
+- name: Query an existing API key by name
+  cisco.nd.nd_api_key:
+    api_key_name: "ansible_test_key"
+    state: query
+  register: query_result
+
+- name: Query all API keys
+  cisco.nd.nd_api_key:
+    state: query
+  register: all_keys
+
+- name: Delete an API key by ID
+  cisco.nd.nd_api_key:
+    api_key_id: "12345678-1234-1234-1234-1234567890ab"
+    state: absent
+
+- name: Delete an API key by name
+  cisco.nd.nd_api_key:
+    api_key_name: "ansible_test_key"
+    state: absent
+
+- name: Update API key annotations
+  cisco.nd.nd_api_key:
+    api_key_name: "existing_key"
+    annotations:
+      owner: "updated_owner"
+      environment: "production"
+    state: present
+"""
+
+RETURN = r"""
+current:
+  description: The current state of the API key after the task has been performed.
+  returned: always
+  type: dict
+  sample:
+    apiKeyName: "ansible_test_key"
+    id: "12345678-1234-1234-1234-1234567890ab"
+    key: "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9..."
+    annotations:
+      owner: "ansible"
+      purpose: "automation"
+    createdTime: "2025-01-01T12:00:00.000Z"
+    lastUsedTime: "2025-01-01T12:30:00.000Z"
+
+previous:
+  description: The previous state of the API key before the task was performed.
+  returned: when state is changed
+  type: dict
+  sample:
+    apiKeyName: "ansible_test_key"
+    id: "12345678-1234-1234-1234-1234567890ab"
+    annotations:
+      owner: "old_owner"
+
+proposed:
+  description: The proposed configuration that would be applied.
+  returned: when check_mode is enabled
+  type: dict
+  sample:
+    apiKeyName: "ansible_test_key"
+    annotations:
+      owner: "ansible"
+      purpose: "automation"
+"""
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        api_key_id=dict(type="str"),
+        api_key_name=dict(type="str", aliases=["description"]),
+        annotations=dict(type="dict", default={}),
+        state=dict(type="str", default="present", choices=["present", "absent", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["api_key_name"]],
+            ["state", "absent", ["api_key_name", "api_key_id"], True],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    api_key_id = nd.params.get("api_key_id")
+    api_key_name = nd.params.get("api_key_name")
+    annotations = nd.params.get("annotations")
+    state = nd.params.get("state")
+
+    path = "/api/v1/infra/aaa/apiKeys"
+    nd.stdout = ""
+    if api_key_id:
+        query_path = "{0}/{1}".format(path, api_key_id)
+        nd.existing = nd.previous = deepcopy(nd.query_obj(query_path))
```

Comment:
since it is not used anywhere else don't think we have to assign it to a variable
```suggestion
        nd.existing = nd.previous = deepcopy(nd.query_obj("{0}/{1}".format(path, api_key_id)))
```